src/http/api/views/account.view.ts

```diff
@@ -0,0 +1,331 @@
+                ),
+                this.db.raw(
+                    '(select count(*) from follows where follows.following_id = accounts.id) as follower_count',
+                ),
+            )
+            .first();
+
+        if (!accountData) {
+            return this.viewByApIdRemote(apId, context);
+        }
+
+        let followedByMe = false;
+        let followsMe = false;
+
+        if (context.site) {
+            const defaultSiteAccount = await this.db('accounts')
+                .innerJoin('users', 'users.account_id', 'accounts.id')
+                .innerJoin('sites', 'sites.id', 'users.site_id')
+                .where('sites.id', context.site.id)
+                .select('accounts.id')
+                .first();
+
+            if (defaultSiteAccount) {
+                followedByMe =
+                    (
+                        await this.db('follows')
+                            .where('follower_id', defaultSiteAccount.id)
+                            .where('following_id', accountData.id)
+                            .first()
+                    )?.id !== undefined;
+
+                followsMe =
+                    (
+                        await this.db('follows')
+                            .where('follower_id', accountData.id)
+                            .where('following_id', defaultSiteAccount.id)
+                            .first()
+                    )?.id !== undefined;
+            }
+        }
+
+        return {
+            id: accountData.id,
+            name: accountData.name,
+            handle: getAccountHandle(
+                new URL(accountData.ap_id).host,
+                accountData.username,
+            ),
+            bio: sanitizeHtml(accountData.bio || ''),
+            url: accountData.url,
+            avatarUrl: accountData.avatar_url || '',
+            bannerImageUrl: accountData.banner_image_url || '',
+            customFields: {},
+            postCount: accountData.post_count,
+            likedCount: accountData.like_count,
+            followingCount: accountData.following_count,
+            followerCount: accountData.follower_count,
+            followedByMe,
+            followsMe,
+            attachment: [],
+        };
+    }
+
+    /**
+     * View an account by its AP ID
+     *
+     * This will attempt to resolve the account via the Fediverse
+     *
+     * @param apId AP ID
+     * @param context View context
+     */
+    private async viewByApIdRemote(
+        apId: string,
+        context: ViewContext,
+    ): Promise<AccountDTO | null> {
+        const fedifyContext = this.fedifyContextFactory.getFedifyContext();
+
+        const actor = await lookupObject(fedifyContext, apId);
+
+        if (actor === null) {
+            throw new Error(`Could not find Actor ${apId}`);
+        }
+
+        if (!isActor(actor)) {
+            return null;
+        }
```

Comment:
I think this might just be a refactor/following some existing pattern - but I still think it's good to confirm, and change if we need to.

Why do we throw if `actor` is null, but we return `null` if it's not an actor? In this context I think I would expect a 404 in both cases 🤔

Reply:
Yeh that was existing code, but I agree, in both cases it should 404

Reply:
updated in https://github.com/TryGhost/ActivityPub/pull/505/commits/e67df95aeebd6d4458b1f832baeeaf0fbee18e45